credentials/apps/credentials/utils.py

```
@@ -53,41 +56,109 @@ def get_programs():
         return cached
 
     programs_api = get_program_api_client()
-    if programs_api is None:
-        return no_programs
-
-    try:
-        response = programs_api.programs.get()
-    except Exception:  # pylint: disable=broad-except
-        log.exception('Failed to retrieve programs from the Programs API.')
-        return no_programs
-
+    response = programs_api.programs.get()
     results = response.get('results', no_programs)
```

If the response is missing the results key, there is probably a bigger issue at hand. I'm inclined to let this bubble up and render an error, rather than hide the fact that the API response is invalid.